docker/mongodb-agent/Dockerfile

```diff
@@ -1,5 +1,40 @@
-ARG imagebase
-FROM ${imagebase} as base
+# the init database image gets supplied by pipeline.py and corresponds to the operator version we want to release
+# the agent with. This enables us to release the agent for older operator.
+ARG init_database_image
+FROM ${init_database_image} AS init_database
+
+FROM public.ecr.aws/docker/library/golang:1.24 AS dependency_downloader
+
+WORKDIR /go/src/github.com/mongodb/mongodb-kubernetes/
+
+COPY go.mod go.sum ./
+
+RUN go mod download
+
+FROM public.ecr.aws/docker/library/golang:1.24 AS readiness_builder
+
+WORKDIR /go/src/github.com/mongodb/mongodb-kubernetes/
+
+COPY --from=dependency_downloader /go/pkg /go/pkg
+COPY . /go/src/github.com/mongodb/mongodb-kubernetes
+
+RUN CGO_ENABLED=0 GOFLAGS=-buildvcs=false go build -o /readinessprobe ./mongodb-community-operator/cmd/readiness/main.go
+RUN CGO_ENABLED=0 GOFLAGS=-buildvcs=false go build -o /version-upgrade-hook ./mongodb-community-operator/cmd/versionhook/main.go
```

Comment:
nit: should we have these two `go build` commands in just one `RUN` statement? That would result into less layers and potentially smaller image.

Reply:
I agree, but in this PR we are not optimising the Dockerfiles. We want to make sure we don't break anything and have exactly the same Dockerfiles as we used before. We should optimise them definitely, but a little bit later